atomic_reactor/util.py

```diff
@@ -738,21 +747,45 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
 
         # Only compare prefix as response may use +prettyjws suffix
         # which is the case for signed manifest
-        response_h_prefix = response.headers['Content-Type'].rsplit('+', 1)[0]
+        try:
+            received_media_type = response.headers.get('Content-Type')
+        except KeyError:
+            # Guess content_type from contents
+            try:
+                encoding = guess_json_utf(response.content)
+                manifest = json.loads(response.content.decode(encoding))
+                received_media_type = manifest['mediaType']
+            except (ValueError,  # not valid JSON
+                    KeyError):   # no mediaType key
+                logger.warning("Unable to fetch media type: neither Content-Type header "
+                               "nor mediaType in output was found")
```

Comment:
Its covered now, added a special case to `test_pulp_pull` to send a broken response